client/asset/btc/btc.go

```diff
@@ -1224,31 +1228,34 @@ func (btc *ExchangeWallet) sendWithReturn(baseTx *wire.MsgTx, addr btcutil.Addre
 		return nil, nil, fmt.Errorf("not enough funds to cover minimum fee rate. %d < %d",
 			totalIn, minFee+totalOut)
 	}
+
 	changeScript, err := txscript.PayToAddrScript(addr)
 	if err != nil {
 		return nil, nil, fmt.Errorf("error creating change script: %v", err)
 	}
 	changeIdx := len(baseTx.TxOut)
 	changeOutput := wire.NewTxOut(int64(remaining-minFee), changeScript)
-	var changeAdded bool
-	isDust := dexbtc.IsDust(changeOutput, feeRate)
+	changeAdded := !dexbtc.IsDust(changeOutput, feeRate)
 	sigCycles := 1
-	if !isDust {
-		// Add the change output with a recalculated fees
-		size += dexbtc.P2WPKHOutputSize
+	if changeAdded {
+		// Add the change output.
+		size0 := baseTx.SerializeSize()
+		baseTx.AddTxOut(changeOutput)
+		changeSize := baseTx.SerializeSize() - size0 // may be dexbtc.P2WPKHOutputSize
+		btc.log.Debugf("Change output size = %d, addr = %s", changeSize, addr.String())
+
+		size += changeSize
 		fee := feeRate * uint64(size)
 		changeOutput.Value = int64(remaining - fee)
-		baseTx.AddTxOut(changeOutput)
-		changeAdded = true
+
 		// Find the best fee rate by closing in on it in a loop.
-		tried := map[uint64]uint8{fee: 1}
+		tried := map[uint64]bool{fee: true}
 		for {
-			sigCycles++
 			msgTx, err = btc.wallet.SignTx(baseTx)
 			if err != nil {
 				return nil, nil, fmt.Errorf("signing error: %v", err)
 			}
-			size := msgTx.SerializeSize()
+			size = msgTx.SerializeSize() // this could be different because...
```

Comment:
The signature size(s) might be different now that the transaction serialization has changed (with the addition of a change output).

Reply:
The only change is passing through SignTx here though.  The change output is added and the size computed above `size += changeSize`

Reply:
But when it goes through `SignTx`, the new signatures could be a different size than the old ones.